src/ir/auth_logic/ast.h

```diff
@@ -53,6 +53,14 @@ class Predicate {
   const std::vector<std::string>& args() const { return args_; }
   Sign sign() const { return sign_; }
 
+  // < operator is needed for btree_set, which is only used for declarations.
+  // Since declarations are uniquely defined by the name of the predicate,
+  // this implementation that just uses < on the predicate names should be
+  // sufficent in the context where it is used.
+  bool operator<(const Predicate& otherPredicate) const {
+    return this->name() < otherPredicate.name();
+  }
```

Comment:
This is a perfectly fine implementation of `operator<`. but why do you need order on predicates? Why not just use an unordered `flat_hash_set`?

Reply:
see other thread about `btree_set` and order of things.

Reply:
resolving because it's back to `flat_hash_set`